src/main/java/softdesign/Robot.java

```diff
@@ -104,21 +101,28 @@ private void ensureNeighbouringImagesTaken() {
 		takeImageIfNeeded(currentDirection.rightBy(3), leftCamera);
 	}
 
+	private Map getMap() {
+		return centralStation.getMap();
+	}
+
 	private void takeImageIfNeeded(Direction direction, CameraSensor camera) {
-		CartesianCoordinate coordinate = tileAhead(direction);
+		CartesianCoordinate coordinate = tileAhead(direction, 1);
+
+		if(getMap().getTile(coordinate) == Map.Tile.EMPTY) {
+			BufferedImage image = camera.createCompatibleImage();
+			centralStation.saveImage(image);
```

Comment:
This is too primitive as it assumes the robot can instantly beam the image to the central station. Please see one of the issues regarding this.

Reply:
Ill look into this, but all the communication id like to do in a separate branch

Reply:
Remember that we are not graded for code quality. I am willing to fix things which require little effort or are important for our grades, but not everything. Hope you understand